### DISCUSSION TITLE
Support compressed depth format: "32FC1; compressedDepth png"

### DISCUSSION THREAD

**Discussion by lucasw** [foxglove/community#212] [Q&A]
This is the compressed format of a 32FC1 depth image (the Zed camera for example outputs floating point).

Currently this is shown in at least the docker version of foxglove:
```
Error: The source image could not be decoded.
```

**Comment by defunctzombie** [comment]
It looks like we already have support for this image format:
https://github.com/foxglove/studio/blob/main/packages/studio-base/src/panels/ImageView/renderImage.ts#L172-L173

Can you provide a sample message that is not working? Please include the entire message, schema, and datatype.

**Comment by lucasw** [comment]
I don't know typescript but I'm guessing it is doing an exact match on `32FC1`, not a partial?

The format as it appears in the message, not just `32FC1` by itself:

```
 format: "32FC1; compressedDepth png"
```

The source code https://github.com/ros-perception/image_transport_plugins/blob/2ec945bb28f919f195f6f4b10e67dc079535ff85/compressed_depth_image_transport/src/codec.cpp#L73-L90

That splits on the `;` and then matches on `compressedDepth png`.

I'll have to double check but this should produce the format from an uncompressed 32FC1

```
rosrun image_transport republish raw in:=/depth_registered compressedDepth out:=/depth_compressed
```

**Comment by defunctzombie** [comment]
It looks like the message type here is `sensor_msgs/CompressedImage`.

From the docs on `sensor_msgs/CompressedImage` https://docs.ros.org/en/api/sensor_msgs/html/msg/CompressedImage.html it doesn't look like thats a valid `format` value. Is this type of format value (with the `;` separator) documented as a common practice for image transports? Could you link us to where that's described or documented?

**Comment by lucasw** [comment]
No it isn't common or documented well (or at all)- maybe there is more beyond that source code in `image_transport_plugins` but I haven't found it.  The git blames show the `';'` split going back many years though.

**Comment by lucasw** [comment]
I found out later that the format is actually 16-bit png, but with a special header with parameters to scale it into floating point- instead of a linear conversion (like storing mm as int then fixed scale factor into meters) it enables more resolution closer to the sensor and less further away:

https://answers.ros.org/question/407659/how-to-generate-format-32fc1-compresseddepth-png-image-in-python-from-float-array/
